### PR TITLE
CenterList Enhance #390 resolved

### PR DESCRIPTION
**Pull request by mayank-kgp**
Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Apply the `MifosStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.
commented code removed

### CODE REVIEW

**Comment by therajanmaurya** [comment]
@mayank-kgp Can you please put screenshot here and Please follow the [Commit Style Guide](https://github.com/openMF/android-client/wiki/Commit-Style-Guide).

**Comment by mayank-kgp** [comment]
The new client screen after these changes would appear as - 

centerlist

**Comment by mayank-kgp** [comment]
@therajanmaurya  Please have a look at this.

**Comment by mayank-kgp** [comment]
@therajanmaurya Do I need to change anything else or is it proper?

**Comment by mayank-kgp** [comment]
@therajanmaurya Please let me know if any further information has to be added in UI or it is proper for now.

**Comment by mayank-kgp** [comment]
@therajanmaurya Can you please help me in information which should be shown?
We are getting the information - https://demo.openmf.org/fineract-provider/api/v1/centers?tenantIdentifier=default

**Comment by mayank-kgp** [comment]
@therajanmaurya  I am adding new screenshot. Please have a look.

new scr

**Comment by therajanmaurya** [comment]
@mayank-kgp Follow this Mock up 😊

image

**Comment by mayank-kgp** [comment]
@therajanmaurya I have done the work. Corresponding screenshot is attached below.
This is my first PR wherein I have worked related to UI. Please review it. 

centers

**Comment by therajanmaurya** [comment]
image

**Comment by mayank-kgp** [comment]
center2

**Comment by ishan1604** [comment]
@mayank-kgp You should not have text size any smaller than `12sp` in your apps. 

Also Consider aligning the green/red dot center vertically of the row item :)

**Comment by mayank-kgp** [comment]
@ishan1604 Yeah, I was getting the same warning in android studio for text size.
Please see the screenshot with 12 sp and gravity of dot is center_vertical .  :)

center4

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/adapters/CentersListAdapter.java [hunk 1]
```diff
@@ -89,27 +112,33 @@ public int getItemCount() {
...
     public static class ViewHolder extends RecyclerView.ViewHolder {
 
+        @BindView(R.id.iv_status_indicator)
+        CircularImageView iv_status_indicator;
+
+        @BindView(R.id.tv_account_number)
+        TextView tvAccountNumber;
+
         @BindView(R.id.tv_center_name)
         TextView tv_center_name;
```
Comment (therajanmaurya) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/adapters/CentersListAdapter.java]: Change name to tvCenterName

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/adapters/CentersListAdapter.java [hunk 2]
```diff
@@ -46,21 +55,35 @@ public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
     @Override
     public void onBindViewHolder(ViewHolder holder, int position) {
 
-        String centerId = context.getResources()
-                .getString(R.string.center_id) + centers.get(position).getId();
+        centerId = "" + centers.get(position).getId();
```
Comment (therajanmaurya) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/adapters/CentersListAdapter.java]: It should be like this.
```java
@Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        Center center = centers.get(position);
 
        holder.tvAccountNumber.setText("Account No: " + String.valueOf(center.getAccountNo()));
        holder.tvCenterId.setText(String.valueOf(center.getId()));
        holder.tv_center_name.setText(center.getName());
        if (center.getStaffId() != null) {
           holder.tvStaffId.setText(String.valueOf(center.getStaffId()));
           holder.tvStaffName.setText(center.getStaffName());
        } else {
            staffId = "";
            holder.tvStaffName.setText("no staff");
        }
        holder.tvOfficeId.setText(String.valueOf(center.getOfficeId()));
        holder.tvOfficeName.setText(center.getOfficeName());
        if (center.getActive()) {
            holder.iv_status_indicator.setImageDrawable(
                    Utils.setCircularBackground(R.color.light_green, context));
        } else {
            holder.iv_status_indicator.setImageDrawable(
                    Utils.setCircularBackground(R.color.light_red, context));
        }
    }
```
**Note** Please include Hard Coded String in strings.xml

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/adapters/CentersListAdapter.java [hunk 3]
```diff
@@ -89,27 +112,33 @@ public int getItemCount() {
 
     public static class ViewHolder extends RecyclerView.ViewHolder {
 
+        @BindView(R.id.iv_status_indicator)
+        CircularImageView iv_status_indicator;
```
Comment (therajanmaurya) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/adapters/CentersListAdapter.java]: Change to ivStatusIndicator

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/adapters/CentersListAdapter.java [hunk 4]
```diff
@@ -89,27 +112,33 @@ public int getItemCount() {
...
 
         @BindView(R.id.tv_staff_name)
-        TextView tv_staff_name;
+        TextView tvStaffName;
 
         @BindView(R.id.tv_staff_id)
-        TextView tv_staff_id;
-
+        TextView tvStaffId;
+        //
```
Comment (therajanmaurya) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/adapters/CentersListAdapter.java]: Remove this comment

**Comment by mayank-kgp** [comment]
@therajanmaurya Please review it.

File: mifosng-android/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -543,6 +543,8 @@
     </string-array>
 
     <string name="surveyEmpty">This Survey is Empty</string>
+    <string name="centerList_account_prefix">Account No: %1$s. </string>
+    <string name="centerList_no_staff">no staff</string>
```
Comment (therajanmaurya) [inline mifosng-android/src/main/res/values/strings.xml]: Do not need to add prefix ```centerList``` in ```centerList_no_staff```, So stay with ```no_staff```

**Comment by mayank-kgp** [comment]
@therajanmaurya  done

**Comment by therajanmaurya** [comment]
@ishan1604 give your final review and merge.

**Comment by ishan1604** [comment]
@mayank-kgp this is the last time I merged your commit with bad messages.

Please don't add rubbish jargon in your commit messages when you are sending Pull Requests.

**Comment by mayank-kgp** [comment]
@ishan1604  I will take care from next time.
I thought that only the main commit message is important after squashing. I didn't care about commit messages that are being squashed into the main commit.
This will not happen from now.
Thanks for merging. :)

**Pull request merged**